### PR TITLE
Fix #3031: Resolved glide related bug that was shrinking the image

### LINKED ISSUE #3031: Glide loading small image
**Describe the bug**
Earlier, we use to load drawable using `setImageResource`. Shifting to Glide, causes a bug of smaller image size in Topic Revision Screen only

**To Reproduce**
1. Click on Fractions card on Home Screen
2. Go to Revision Tab

**Expected behavior**
| Currently | What it should be |
| -- | -- |
|  |  |

Look into `LessonThumbnailImageView` on how to load a drawable. 

https://github.com/oppia/oppia-android/blob/1be8ccde14c25d763c10dd8902ec7f5a2fa6f99e/app/src/main/java/org/oppia/android/app/customview/LessonThumbnailImageView.kt#L86

**Additional context**
The commit which cause this bug - https://github.com/oppia/oppia-android/commit/2bcd5114eb614e06f590ec759fde4b6bf28149ce#diff-a37faf8490754f9ab5367852640b3631eea31a197d74a0d5b8dbc8a0ad5e1c6e

**How to send a Pull Request**
1. Run `TopicRevisionFragmentTest` test on Espresso
2. Add a screenshot in Pull Request description showing all the tests are passing

### PR DESCRIPTION
**Pull request by jonathanalvares9009**
## Explanation

This PR will resolve a bug associated with the glide library that was shriking the size of the image.
Fixes #3031 
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by jonathanalvares9009** [comment]
**Before this pull request:**


**After this pull request:**

**Comment by jonathanalvares9009** [comment]
@anandwana001 can you review my PR?

File: utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt
Comment (anandwana001) [inline utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt]: Could you please be more descriptive on this why we don't need transformation?

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Please check the How to send pull request section in the issue.

File: utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt
Comment (jonathanalvares9009) [reply utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt]: @anandwana001 since the images have already been pre-scaled

**Comment by jonathanalvares9009** [comment]
@anandwana001 the expresso tests are failing before my change and same tests are failing after my change

File: utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt
Comment (anandwana001) [reply utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt]: Could you check does this change affecting any other part of the app or not?

**Comment by FareesHussain** [comment]
@anandwana001 According to this https://github.com/oppia/oppia-android/pull/3025#issuecomment-811498277 
The failing tests are not dependent on the image size.
These tests are fixed in #3025 for both Robolectric and Espresso (https://github.com/oppia/oppia-android/pull/3025#issuecomment-811496635)

But we do need Original Image size so I guess we don't require tests here

**Comment by anandwana001** [comment]
Sounds good to me, thanks for the update too. 
@jonathanalvares9009 Could you update the branch with the latest develop as the tests are fixed now, only thing we need to fix here is the image size.

File: utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt
Comment (jonathanalvares9009) [reply utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt]: @anandwana001 it doesn't affect anything

**Comment by jonathanalvares9009** [comment]
Screenshot 2021-04-01 at 4 00 02 PM

@anandwana001 it passes all tests. N thanks @FareesHussain

File: utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt
Comment (anandwana001) [reply utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt]: Could you confirm the story screen, go to lesson click on topic, it will open a story activity with a list of cards, and check if any card is blur or not!!

**Comment by jonathanalvares9009** [comment]
@anandwana001 and @BenHenning can you review my PR?

**Comment by anandwana001** [comment]
PTAL - https://github.com/oppia/oppia-android/pull/3032#discussion_r605770579

File: utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt
Comment (jonathanalvares9009) [reply utility/src/main/java/org/oppia/android/util/parser/GlideImageLoader.kt]: @anandwana001 I checked it on an emulator (pixel 5 API 29) and realme 5 (API 30) I found no new issues created.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
In the first place, if we don't transform we cannot take the advantage of `ImageTransformation` which is used to blur the image at some of the cards. The commit I had mentioned in the issue is introducing this blur feature. 

If you see the screenshot below, the image use to blur, but in this PR it's not because we are not transforming. 

Screenshot_1617417024

What do we need in order to solve this issue first look for the reason why the image is getting small, then we can look for solution.

**Comment by jonathanalvares9009** [comment]
@anandwana001 I'll look into the issue.

**Comment by jonathanalvares9009** [comment]
@anandwana001 I removed dontTransform. But instead of using centerInside I used fitCenter. centerInside reduces the image size to fit it in the center whereas fitCenter maintains the aspect ratio

**Comment by jonathanalvares9009** [comment]
Screenshot 2021-04-03 at 1 21 24 PM

@anandwana001 passed all tests and it resolves the bug without affecting anything

**Comment by anandwana001** [comment]
Could you please also add 2 screenshots, one of revision tab and other of the story activity, same screen I added above?

**Comment by jonathanalvares9009** [comment]
@anandwana001 PTAL I added the screenshots

**Comment by anandwana001** [comment]
@jonathanalvares9009 It looks good to me. Before approving could you confirm that other parts of the app where the images are loading have no issues, like explorations. You can check by playing lessons in the lessons tab and also the practice tab.

**Comment by jonathanalvares9009** [comment]
@anandwana001 I checked the app, it has no issues.

**Comment by jonathanalvares9009** [comment]
@anandwana001 can you review this?

**Review by anandwana001 — APPROVED** [review]
LGTM, defer to @rt4914 if anything seems missing.

**Comment by jonathanalvares9009** [comment]
@rt4914 can you review my PR?

**Comment by anandwana001** [comment]
@rt4914 as per the alpha content, I think this pr might solve - https://github.com/oppia/oppia-android/issues/2962

**Comment by rt4914** [comment]
@jonathanalvares9009 and @anandwana001 At what all places we have seen this issue?
The current implementation solves it for `Topic-Revision-Tab` only and not anywhere else and therefore it won't solve #2962

**Comment by jonathanalvares9009** [comment]
@rt4914 can you review this?

**Comment by anandwana001** [comment]
@rt4914 Till now we faced this issue in the practice tab only, and talking about the #2962 which has alpha content. So, I guess we need to review this PR with the alpha content to check if this pr solves it or not.

**Comment by rt4914** [comment]
Edit:
Lets check this on alpha content first.

**Comment by rt4914** [comment]
> Edit:
> Lets check this on alpha content first.

@BenHenning We want to check this PR on alpha content so how can I do that? Do you have alpha files downloaded somewhere?

**Comment by BenHenning** [comment]
@rt4914 will follow up via email.

**Comment by rt4914** [comment]
@jonathanalvares9009 Will need some more time reviewing this PR. Meanwhile you can work on the issue that I shared with you today.

**Comment by jonathanalvares9009** [comment]
Ok @rt4914

**Review by rt4914 — APPROVED** [review]
I have checked this on alpha content.
This works nicely.

**Review by rt4914 — CHANGES_REQUESTED** [review]
I missed one thing. Apply this change to other variations too. `layout-land`, `layout-sw600dp`, etc.

**Comment by jonathanalvares9009** [comment]
@rt4914 Made the requested changes

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @jonathanalvares9009. Could you please add tests to verify that the property is correctly set in XML for each of the different situations (e.g. portrait, landscape, and the sw600dp versions of each)? You should be able to add tests to TopicRevisionFragmentTest that set up the different environment via qualifiers to verify each property is set (an easy check is that removing the fix for one XML file should correspond to a single test failing for that case).

@rt4914 thanks for verifying this with production data. Is it the case that #2962 will require a different solution? From what I can tell those thumbnails already use fitCenter.

**Comment by rt4914** [comment]
> Thanks @jonathanalvares9009. Could you please add tests to verify that the property is correctly set in XML for each of the different situations (e.g. portrait, landscape, and the sw600dp versions of each)? You should be able to add tests to TopicRevisionFragmentTest that set up the different environment via qualifiers to verify each property is set (an easy check is that removing the fix for one XML file should correspond to a single test failing for that case).
> 
> @rt4914 thanks for verifying this with production data. Is it the case that #2962 will require a different solution? From what I can tell those thumbnails already use fitCenter.

@BenHenning Mostly yes. Given that I have done alpha setup on my other PC I will be testing all these alpha bugs on that so that we can finish all these minor fixes. Mostly this will work for #2962 but I will check this correctly on alpha app and then create a PR.

**Comment by rt4914** [comment]
@BenHenning Suggested approach to fix this fully.

1. Inside `LessonThumbnailImageView` we set the scale type to `FIT_CENTER`. This will help us to avoid any UI regression even if it is changed in xml file. 
2. Remove `scaleType` for `LessonThumbnailImageView` in all xml files. 
3. Write tests in all files which uses `LessonThumbnailImageView` to ensure that the `scaleType` is `FIT_CENTER`.

Output on alpha:

**Comment by BenHenning** [comment]
That approach LGTM @rt4914 & is a really nice way to consolidate.

**Comment by rt4914** [comment]
@jonathanalvares9009 Can you apply changes as per this comment: https://github.com/oppia/oppia-android/pull/3032#issuecomment-825560947

**Comment by jonathanalvares9009** [comment]
Ok @rt4914 will do it

**Comment by jonathanalvares9009** [comment]
Hi @rt4914 I'm having trouble writing the tests can you help me?

**Comment by anandwana001** [comment]
@jonathanalvares9009 any update on this?

**Comment by anandwana001** [comment]
Closing PR as no further update.

**Pull request closed without merging**
